---
layout: post
title:  "Apache 서버에 Django 앱 배포 시 'Authentication credentials were not provided' 에러"
date:   2018-01-11 14:48:00 +0900
categories: back-end django
---

로컬에서 잘 동작하던 Django 앱이 Apache 환경에 배포하자 마자 아래와 같은 에러가 발생하면서 정상 동작하지 않기 시작했다.

> Authentication credentials were not provided

분명 로컬에서는 아무 문제 없었는데! 라면 Apache가 원인일 가능성이 크다. 별도의 설정을 하지 않으면 Apache가 Django의 인증 관련 헤더를 날려버린다고 한다. 아래 설정을 파일에 추가해주자.

    WSGIPassAuthorization On

### 환경
- Ubuntu 14.04
- Apache2
- Python 3.x
- 로컬 환경에서 정상적으로 동작하는 Django 프로젝트

---

### References
- [Django Rest Framework - Authentication credentials were not provided](https://stackoverflow.com/questions/26906630/django-rest-framework-authentication-credentials-were-not-provided)